src/Tests/Microsoft.NET.Publish.Tests/RuntimeIdentifiersTests.cs

```diff
@@ -194,6 +195,81 @@ public void PublishWithRuntimeIdentifier(bool publishNoBuild)
             }
         }
 
+        [WindowsOnlyTheory]
+        [InlineData(false, false)] // publish rid overrides rid in project file if publishing
+        [InlineData(true, false)] // publish rid doesnt override global rid
+        [InlineData(true, true)] // publish rid doesnt override global rid, even if global
+        public void PublishRuntimeIdentifierSetsRuntimeIdentifierAndDoesOrDoesntOverrideRID(bool runtimeIdentifierIsGlobal, bool publishRuntimeIdentifierIsGlobal)
+        {
+            string tfm = ToolsetInfo.CurrentTargetFramework;
+            string publishRuntimeIdentifier = "win-x64";
+            string runtimeIdentifier = "win-x86";
+
+            var testProject = new TestProject()
+            {
+                IsExe = true,
+                TargetFrameworks = tfm
+            };
+            if (!publishRuntimeIdentifierIsGlobal)
+                testProject.AdditionalProperties["PublishRuntimeIdentifier"] = publishRuntimeIdentifier;
+            if (!runtimeIdentifierIsGlobal)
+                testProject.AdditionalProperties["RuntimeIdentifier"] = runtimeIdentifier;
+            testProject.RecordProperties("RuntimeIdentifier");
+
+            List<string> args = new List<string>
+            {
+                runtimeIdentifierIsGlobal ? $"/p:RuntimeIdentifier={runtimeIdentifier}" : "",
+                publishRuntimeIdentifierIsGlobal ? $"/p:PublishRuntimeIdentifier={publishRuntimeIdentifier}" : ""
+            };
+
+            string identifier = $"PublishRuntimeIdentifierOverrides-{publishRuntimeIdentifierIsGlobal}-{runtimeIdentifierIsGlobal}";
+            var testAsset = _testAssetsManager.CreateTestProject(testProject, identifier: identifier);
+            var publishCommand = new DotnetPublishCommand(Log);
+            publishCommand
+                .WithWorkingDirectory(Path.Combine(testAsset.TestRoot, MethodBase.GetCurrentMethod().Name))
+                .Execute(args.ToArray())
+                .Should()
+                .Pass();
+
+            string expectedRid = runtimeIdentifierIsGlobal ? runtimeIdentifier : publishRuntimeIdentifier;
+            var properties = testProject.GetPropertyValues(testAsset.TestRoot, targetFramework: tfm);
+            var finalRid = properties["RuntimeIdentifier"];
+
+            Assert.True(finalRid == expectedRid);
+        }
+
+        [WindowsOnlyFact]
```

Comment:
This test can probably also run on all operating systems.  You might need to choose a `publishRid` that isn't going to be the default RID anywhere.

Reply:
Hm, while we could use something obscure and specific, like `rocky-arm64`, I don't think there's much added value in running this test on more than one platform for the potential ways this can go wrong.